scripts/linters/pylint_extensions.py

```diff
@@ -1960,6 +2059,427 @@ def process_tokens(self, tokens):
+                is called, Possible values can be 'Any', 'object' and 'cast'.
+            line_num: int. The line number where error is encountered.
+            node: astroid.scoped_nodes.Module. Node to access module content.
+        """
+        if exceptional_type == 'Any':
+            self.add_message(
+                'any-type-used', line=line_num, node=node
+            )
+        if exceptional_type == 'object':
+            self.add_message(
+                'object-class-used', line=line_num, node=node
+            )
+        if exceptional_type == 'cast':
+            self.add_message(
+                'cast-func-used', line=line_num, node=node
+            )
+
+    def check_comment_is_present_with_object_class(self, tokens, node):
+        """Checks whether the object class in a module has been documented
+        or not. If the object class is not documented then adds an error
+        message.
+
+        Args:
+            tokens: Token. Object to access all tokens of a module.
+            node: astroid.scoped_nodes.Module. Node to access module content.
+        """
+        object_class_status_dict = copy.deepcopy(
+            self.EXCEPTIONAL_TYPE_STATUS_DICT
+        )
+
+        object_type_regex = r'Here we use object because'
+
+        for (token_type, token, (line_num, _), _, line) in tokens:
+            line = line.strip()
+
+            if token_type == tokenize.COMMENT:
+                if re.search(object_type_regex, line):
+                    object_class_status_dict['type_comment_present'] = True
+                    object_class_status_dict['type_comment_line_num'] = line_num
+
+            self._check_exceptional_type_is_documented(
+                object_class_status_dict, None, token_type, token,
+                line, line_num, 'object', node
+            )
+
+    def check_comment_is_present_with_cast_method(self, tokens, node):
+        """Checks whether the cast method in a module has been documented
+        or not. If the cast method is not documented then adds an error
+        message.
+
+        Args:
+            tokens: Token. Object to access all tokens of a module.
+            node: astroid.scoped_nodes.Module. Node to access module content.
+        """
+        cast_type_regex = r'Here we use cast because'
+        cast_comment_present = False
+        cast_comment_line_num = 0
+        import_status_dict = {
+            'single_line_import': False,
+            'import_line_num': 0,
+            'inside_multi_line_import_scope': False
+        }
+
+        for (token_type, token, (line_num, _), _, line) in tokens:
+            line = line.strip()
+
+            if token_type == tokenize.COMMENT:
+                if re.search(cast_type_regex, line):
+                    cast_comment_present = True
+                    cast_comment_line_num = line_num
+
+            self._check_import_status(
+                import_status_dict, token_type, token, line_num
+            )
```

Comment:
Instead of checking whether you're currently inside an import, is it possible to check whether `cast` is being called? That seems simpler. e.g. we care about instances of `cast(...)` but not instances of just `cast`

Reply:
> is it possible to check whether cast is being called?

Yeah, it's possible but if we only check that cast is being called then how do we check for comments that are present above it?

Currently, we are traversing through token-to-token, and whenever we encounter a comment we keep that info as a flag, and when we encounter cast we use that flag to check if that is a ok case or not. Pleas provide more info if you think I'm missing something.